service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/container/RetrieveDestinationStorageAccountResourceIdStep.java

```diff
@@ -0,0 +1,124 @@
+package bio.terra.workspace.service.resource.controlled.flight.clone.azure.container;
+
+import bio.terra.common.iam.BearerToken;
+import bio.terra.landingzone.db.exception.LandingZoneNotFoundException;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.amalgam.landingzone.azure.LandingZoneApiDispatch;
+import bio.terra.workspace.common.utils.FlightUtils;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.generated.model.ApiAzureLandingZoneDeployedResource;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.resource.controlled.flight.clone.azure.container.exception.InvalidStorageAccountException;
+import bio.terra.workspace.service.resource.model.WsmResourceFamily;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Attempts to fetch the storage account ID for a new container in the destination workspace
+ * following process:
+ *
+ * <p>1. Checks the destination workspace for a single storage account; if more than one is present
+ * we are in an unsupported state and fail out.
+ *
+ * <p>2. If no storage accounts are present, attempts to check the owning Landing Zone. If present,
```

Comment:
Should we prefer the landing zone over the workspace's storage account (given that is the direction we are headed)?

Reply:
[Switched](https://github.com/DataBiosphere/terra-workspace-manager/blob/1b725dff35e3e1667f36eb1e7aad95f19269907f/service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/container/RetrieveDestinationStorageAccountResourceIdStep.java#L68-L92) around the landing zone to be the first place we check